api/api.go

```diff
@@ -356,11 +374,27 @@ func (a *API) submitBlock(w http.ResponseWriter, r *http.Request) {
 	switch a.st.ForkVersion(a.st.HeadSlot()) {
 	case structs.ForkCapella:
 		var creq capella.SubmitBlockRequest
-		if err := json.NewDecoder(bytes.NewReader(b)).Decode(&creq); err != nil {
-			a.m.ApiReqCounter.WithLabelValues("submitBlock", "400", "payload decode").Inc()
-			writeError(w, http.StatusBadRequest, errors.New("invalid submitblock request capella decode"))
-			return
+		if r.Header.Get("Content-Type") == "application/octet-stream" {
+			l = l.WithField("requestContentType", "ssz")
+			if err := creq.UnmarshalSSZ(b); err != nil {
+				// Fallback to JSON.
```

Comment:
Content type is octet-stream. I don't think we should fallback to anything here.